src/item_factory.cpp

```diff
@@ -2077,6 +2083,24 @@ void Item_factory::load_basic_info( JsonObject &jo, itype &def, const std::strin
         def.phase = jo.get_enum_value<phase_id>( "phase" );
     }
 
+    JsonObject loadables = jo.get_object( "loadables" );
+    JsonArray integrals = loadables.get_array( "integrals" );
+    for( size_t i = 0; i < integrals.size(); ++i ) {
+        JsonObject integral_def = integrals.get_object( i );
+        itype integral_type;
+        if( integral_def.get_string( "type" ) == "MAGAZINE" ) {
+            load_slot( integral_type.magazine, integral_def, src );
+            load_basic_info( integral_def, integral_type, src );
+            if( integral_type.name == "none" ) {
+                integral_type.name = string_format( "%s integral magazine", jo.get_string( "name" ) );
```

Comment:
Needs translation. Why make the name optional anyway?

Reply:
I was looking at using anonymous items that only have a handful of attributes for integral items but with the way the code is already set up it will work better to do them as proper items.